### ISSUE TITLE
bug(MatError): Screen reader doesn't announce errors shown after tabbing out of the field

### ISSUE THREAD

**Issue by clamli** [angular/components#29616]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version
~~_I lean towards uncheck this checkbox even though it works in the previous version, because I suspect the issue resides in either screen readers or browsers._~~ 

It looks like the cause of the issue is [the MatError logic of handling live region](https://stackoverflow.com/questions/76071408/nvda-not-reading-mat-error-on-firefox), here is [a raw HTML implementation that reflects the same behavior](https://codebin.googleplex.com/#/jhsbbn2npnd).

### Description

Some screen readers don't announce errors shown after tabbing out to the next interactive element.

The screen reader and browser combination that DOES announce the error after tabbing out

- NVDA + Chrome (https://screencast.googleplex.com/cast/NTY2NTM5NjQ0NjE5OTgwOHxiZTAzYzQyYS02Ng)
- JAWS + Chrome
- JAWS + Firefox

The screen reader and browser combination that DOES NOT announce the error after tabbing out

- NVDA + Firefox (https://screencast.googleplex.com/cast/NjEzOTg4NjcxNjMyMTc5Mnw3YmJlNGVlNS0zNw)
- VO + Chrome (https://screencast.googleplex.com/cast/NTc0ODU0OTA5MzQyNTE1MnwzYjNiZmMzNS1hMw)
- VO + Firefox

### Reproduction

https://material.angular.io/components/form-field/overview#form-field-error
Steps to reproduce:
1. Go to the example above
2. Tab to and out of the form field with MatError using different screen reader and browser combinations and observe the announcement 

### Expected Behavior

The error message should always be announced by screen readers.

### Actual Behavior

The error message is announced for NVDA + Chrome, JAWS + Chrome and JAWS + Firefox, but is not announced for NVDA + Firefox, VO + Chrome and VO + Firefox.

### Environment

- Angular: 
- CDK/Material: 18.2.0
- Browser(s): Chrome, Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows, macOS
